src/containers/guest-home-page/signup-form/SignupForm.js

```diff
@@ -28,6 +28,12 @@ const SignupForm = ({ handleSubmit, handleChange, handleBlur, data, errors, clos
     setButtonDisabled(!buttonDisabled)
   }
 
+  const hasErrors = useCallback(() => Object.values(errors).some((elem) => elem === undefined), [errors])
+
+  const emptyData = useCallback(() => Object.values(data).every((elem) => elem !== ''), [data])
```

Comment:
if u need only result and not the function use useMemo instead useCallback

Reply:
yes, i used useCallback in those separete functions. now i refactored, removed separete funcs and putt everything into useMemo